src/app/api/challenge/start/route.ts

```diff
@@ -1,37 +1,50 @@
-import { NextResponse } from 'next/server';
-import { createRedisClient, getChallengeFromCookie } from '@/app/api/utils';
-
-
-export async function POST(request: Request) {
-	const redis = createRedisClient();
-
-	try {
-		const challengeStatus = await getChallengeFromCookie(redis);
-
-		if (!challengeStatus) {
-			return NextResponse.json(
-				{ success: false, message: 'Challenge not found' },
-				{ status: 404 }
-			);
-		}
-
-        // Start the challenge
-        challengeStatus.isStarted = true;
-        challengeStatus.startTime = new Date().getTime(); // Convert to number (timestamp)
-        await redis.set(`challenge:${challengeStatus.emailAddress}`, JSON.stringify(challengeStatus));
-
-        return NextResponse.json(
-            { success: true, message: 'Challenge started' },
-            { status: 200 }
-        );
-
-	} catch (error) {
-		console.error('Error starting challenge:', error);
-		return NextResponse.json(
-			{ success: false, message: 'Internal server error' },
-			{ status: 500 }
-		);
-	} finally {
-		await redis.quit();
-	}
-}
\ No newline at end of file
+import { NextResponse } from "next/server";
+import { createRedisClient, getChallengeFromCookie } from "@/app/api/utils";
+
+export async function POST() {
+  const redis = createRedisClient();
+
+  try {
+    const challengeStatus = await getChallengeFromCookie(redis);
```

Comment:
Do we have to check if the challenge is already started?

Reply:
We are fetching the challenge JSON from the cookies. Renamed the variable 'challengeStatus' to 'challengeDetails'

Reply:
Yes, when a start challenge request is sent we fetch the challenge from cookie. Then we check if it's a valid challenge and has duration and also if it's not started.

If all 3 conditions are satisfied we actually start the challenge